### ISSUE TITLE
[Feature/Bug] Order images first to last when arranged as a grid

### ISSUE THREAD

**Issue by NEKolev** [olive-groves/butterfly_viewer#60]
### Is there an existing request for this feature?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Proper arrangement of the coins in the grid. In my opinion, the expected behaviour is for the images to be arranged from top to bottom, from left to right. In other words, the first image - top left, the second image top right, the third bottom left, the fourth bottom right, and so on. At the moment they are arranged exactly the opposite.

https://github.com/olive-groves/butterfly_viewer/assets/70388317/63163da9-d9e2-42c3-845f-6d1363ae1efb

**Comment by larsmaxfield** [comment]
TL;DR: Good feature, but I probably won't get to it soon because (I think) it's an upstream issue with the interface library I use in the app.
__

Unfortunately the video you added doesn't play for me:

![image](https://github.com/olive-groves/butterfly_viewer/assets/83759569/87487d7b-4179-49ef-a845-a53ad759a129)

Nevertheless if I understand correctly, this is an issue regarding the order of the images when they are arranged as a grid. Specifically, that the order is not in the order they are added into the Viewer (from first to last), but rather in what seems to be reverse order (from last to first).

I agree that the order should be from first added to last added. However, it might be a while before I look into this.

I've actually known of this issue since I started making this app. Unfortunately it is the standard behavior of the `tileSubWindows()` slot in Qt's `QMdiArea`. And it's actually worse than reverse order — rather, the tiling (arranging) order changes based on the last image window you've interacted with, but that change only then happens when you resize the Viewer window.

I remember trying to solve this but giving up. I think because I had successfully implemented both horizontal and vertical arrangement which do indeed preserve the first-to-last image order. 

But now as I write this, perhaps the solution is to simply create my own grid arrangement method which preserves the order. That will be a bit more tricky though, especially when the number of image windows is a prime number above two because one of the windows will need to be bigger.

**Comment by NEKolev** [comment]
> TL;DR: Good feature (and actually more of a bug), but I likely won't get to it soon. __
> 
> Unfortunately the video you added doesn't play for me:
> 
> ![image](https://private-user-images.githubusercontent.com/83759569/338593796-87487d7b-4179-49ef-a845-a53ad759a129.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMjQ5MjIsIm5iZiI6MTcxODEyNDYyMiwicGF0aCI6Ii84Mzc1OTU2OS8zMzg1OTM3OTYtODc0ODdkN2ItNDE3OS00OWVmLWE4NDUtYTUzYWQ3NTlhMTI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE2NTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkMzQ3NTIyN2MwMDI5YjVhYzQzNGU4NmJiMTExZDc2Yjc4MDM4MTBhODgyZTg0ZGZlNWZhZTIwMTJmMzNkNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZPGabdfHYMH_oYlu2OUgTAEEwtcp7DDa60wANFc_no4)
> 
> Nevertheless if I understand correctly, this is an issue regarding the order of the images when they are arranged as a grid. Specifically, that the order is not in the order they are added into the Viewer (from first to last), but rather in what seems to be reverse order (from last to first).
> 
> I agree that the order should be from first added to last added.
> 
> I've actually known of this issue since I started making this app. Unfortunately it is the standard behavior of the `tileSubWindows()` slot in Qt's `QMdiArea`. And it's actually worse than reverse order — rather, the tiling (arranging) order changes based on the last image window you've interacted with, but that change only then happens when you resize the Viewer window. So I would classify it as both a feature and a bug.
> 
> I remember trying to solve this but giving up. I think because I had then successfully implemented both horizontal and vertical arrangement which both preserve the first-to-last image order.
> 
> But now as I write this, perhaps the solution is to simply create my own grid arrangement method which preserves the order. That will be a bit more tricky though, especially when the number of image windows is a prime number above two because one of the windows will need to be bigger.

Most likely you have an ad blocker or something else that stops the playback. I uploaded the clip [here](https://ibb.co/xzpj7W2), but either way, the problem is exactly what you described in detail.

**Comment by larsmaxfield** [comment]
> Most likely you have an ad blocker or something else that stops the playback. I uploaded the clip [here](https://ibb.co/xzpj7W2), but either way, the problem is exactly what you described in detail.

Thanks for confirming and for adding the alternate video link. Indeed the video issue seems to be coming from my side.
